base_layer/wallet/src/test_utils.rs

```diff
@@ -0,0 +1,253 @@
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+use crate::{
+    contacts_service::storage::database::Contact,
+    error::WalletError,
+    storage::{database::WalletBackend, memory_db::WalletMemoryDatabase},
+    wallet::WalletConfig,
+    Wallet,
+};
+use rand::{distributions::Alphanumeric, CryptoRng, OsRng, Rng, RngCore};
+use std::{iter, sync::Arc, thread, time::Duration};
+use tari_comms::{
+    control_service::ControlServiceConfig,
+    peer_manager::{NodeIdentity, PeerFeatures},
+    types::{CommsPublicKey, CommsSecretKey},
+};
+use tari_crypto::{
+    commitment::HomomorphicCommitmentFactory,
+    keys::{PublicKey as PublicKeyTrait, SecretKey as SecretKeyTrait},
+};
+use tari_p2p::initialization::CommsConfig;
+use tari_transactions::{
+    tari_amount::MicroTari,
+    transaction::{OutputFeatures, TransactionInput, UnblindedOutput},
+    types::{PrivateKey, PublicKey, COMMITMENT_FACTORY},
+};
+use tempdir::TempDir;
+use tokio::runtime::Runtime;
+
+// Used to generate test wallet data
+
+pub struct TestParams {
+    pub spend_key: PrivateKey,
+    pub change_key: PrivateKey,
+    pub offset: PrivateKey,
+    pub nonce: PrivateKey,
+    pub public_nonce: PublicKey,
+}
+impl TestParams {
+    pub fn new<R: Rng + CryptoRng>(rng: &mut R) -> TestParams {
+        let r = PrivateKey::random(rng);
+        TestParams {
+            spend_key: PrivateKey::random(rng),
+            change_key: PrivateKey::random(rng),
+            offset: PrivateKey::random(rng),
+            public_nonce: PublicKey::from_secret_key(&r),
+            nonce: r,
+        }
+    }
+}
+pub fn make_input<R: Rng + CryptoRng>(rng: &mut R, val: MicroTari) -> (TransactionInput, UnblindedOutput) {
```

Comment:
Is there a way to not repeat all this at all?

Reply:
So this stuff is usually only used in tests but this is being used to give the Mobile Dev's test data during a proper build of the library. These functions will be temporary (or hidden behind a feature flag or something) but this will come out once we have some infratructure up for wallets to test against. The only other option is to make a test utilities crate elsewhere in the code base?